From 0082e904ee976590e4726a71f8a1e5effb241e90 Mon Sep 17 00:00:00 2001
From: Ross Lagerwall <ross.lagerwall@citrix.com>
Date: Mon, 21 Aug 2017 14:55:39 +0100
Subject: [PATCH] xen: Emit RTC_CHANGE upon TIMEOFFSET ioreq

When the guest writes to the RTC, Xen emulates it and broadcasts a
TIMEOFFSET ioreq. Emit an RTC_CHANGE QMP message when this happens
rather than ignoring it so that something useful can be done with the
information.

Signed-off-by: Ross Lagerwall <ross.lagerwall@citrix.com>
--
Add NULL string to RTC_CHANGE QMP message to match updated API
diff --git a/hw/xen/xen-hvm-common.c b/hw/xen/xen-hvm-common.c
index 78e0bc8f644e..e390a0b7872a 100644
--- a/hw/xen/xen-hvm-common.c
+++ b/hw/xen/xen-hvm-common.c
@@ -12,6 +12,7 @@
 #include "hw/xen/xen-bus.h"
 #include "hw/boards.h"
 #include "hw/xen/arch_hvm.h"
+#include "qapi/qapi-events-misc.h"
 #include "system/runstate.h"
 #include "system/system.h"
 #include "system/xen.h"
@@ -466,6 +467,7 @@ static void handle_ioreq(XenIOState *state, ioreq_t *req)
             cpu_ioreq_move(req);
             break;
         case IOREQ_TYPE_TIMEOFFSET:
+            qapi_event_send_rtc_change((int64_t)req->data, NULL);
             break;
         case IOREQ_TYPE_INVALIDATE:
             xen_invalidate_map_cache();
